src/Workspaces/Core/Portable/Serialization/SerializerService.cs

```diff
@@ -74,11 +74,6 @@ public Checksum CreateChecksum(object value, CancellationToken cancellationToken
         {
             cancellationToken.ThrowIfCancellationRequested();
 
-            if (value is IChecksummedObject checksummedObject)
-            {
-                return checksummedObject.Checksum;
-            }
```

Comment:
existed only to hijack.  but we can already do that throuhg normal virtual methods.